### ISSUE TITLE
Version 14495 blinds are not displayed correctly.

### ISSUE THREAD

**Issue by SargonofAssyria** [domoticz/domoticz#5330]
After upgrade to 14495 (Docker) Blinds are not shown correctly any more.
Blind 'Rechts' is at 20% and blind 'Overloop' is at 30%
Overloop is set to inverted by me to see if it make any difference.
Dimmer lightbulbs show correctly

![02 09 2022_09 58 37_REC](https://user-images.githubusercontent.com/24824113/188088417-89e47b9b-2f8e-446a-a330-62f1c002a689.png)

In my Prod env. (version 14489) with Stas old plugin they show correctly.

**Comment by Thaui57** [comment]
Same with my Qubino shutters. Percentage closed is shown correctly. But totally closed slider is wrong.

**Comment by gizmocuz** [comment]
@SargonofAssyria , could you post your blind config topic+payload and also the payloads when you set your blinds to 30%

Could you also answer the below questions (using normal (not inverted) blinds):
- when you press the open icon (left icon) does the blind open ?
- when you press the close icon (right icon) does the blind clsoe ?
- when you set the slider to 30%, does the blind open 30% (so 70% of the blind is closed)
- what hardware are you using to control the blind ? (fibaro/...._

@Thaui57 , it seems you have a different situation as above as your percentage is correctly but the final open/close does not
Are you using normal (not inverted blinds) (you should do so)
Could you also answer the above questions?

**Comment by SargonofAssyria** [comment]
[TuYa TS0601_cover.txt](https://github.com/domoticz/domoticz/files/9494029/TuYa.TS0601_cover.txt)
Enclosed config of this device: https://www.zigbee2mqtt.io/devices/TS0601_cover.html#tuya-ts0601_cover it actually is a Zemismart AM43
when you press the open icon (left icon) does the blind open ?  YES
when you press the close icon (right icon) does the blind clsoe ?  YES 
when you set the slider to 30%, does the blind open 30% (so 70% of the blind is closed)  YES
The actual display is now correct on my TEST docker 14495 environment.
So it seems to work oke. 
The only problem remaining is that the 'rolgordijn' is controlled by dzVents in my production environment.
All change there are not reflected in the test docker environment, although the zigbee controler send updates that should be reflected in the test GUI. 
It does so in my prod environment.

**Comment by gizmocuz** [comment]
@SargonofAssyria , okey so all seems fine in relation to MQTT-AD and blinds and 14495 ... So this issue can be closed....
But maybe you can change the topic to include dzVents and explain what is wrong

But first, you are using normal blinds and not inverted blinds i hope ?

So when using normal blinds, and you operate the blinds via the GUI in browser 1, and you have another browser open with the event editor where you can see the status overview of the blind, is the status correctly displayed ?

- if you open the blinds, does it say 'open' in the status page ?
- if you close the blinds, does it say 'close' in the status page ?
- if you set the slider to 30%, does it display 30% in the status page ?

If all this is working fine, what happens if:
- you call 'close' from dzVents, does it close the blinds ?
- you call 'open' from dzVents, does it open the blinds ?
- when you set the position to 30%, does it open the blinds for 30% ?
-

**Comment by SargonofAssyria** [comment]
But first, you are using normal blinds and not inverted blinds i hope ? I am using NORMAL blinds.
if you open the blinds, does it say 'open' in the status page ? NO change, not even in the slider.
if you close the blinds, does it say 'close' in the status page ? Change in 'Closed', also the slider goes to the end. See pic.
if you set the slider to 30%, does it display 30% in the status page ? Change in 30% and slider goes to 30%
![06 09 2022_17 08 43_REC](https://user-images.githubusercontent.com/24824113/188670822-eba176ed-2b9e-438a-9417-7facc42a6295.png)


you call 'close' from dzVents, does it close the blinds ? Blinds open, and GUI does NOT change.
you call 'open' from dzVents, does it open the blinds ? Blinds closes, and GUI does NOT change.
when you set the position to 30%, does it open the blinds for 30% ? Blinds go to 70% and GUI does NOT change.

dzVents event messages are correct in the logfile, open, close and setLevel 30%
I use http://192.168.2.YYY:8080/json.htm?type=devices&rid=XXX to get my Status and level values.

**Comment by Thaui57** [comment]
@gizmocuz, I will need the inverted blind functionality because of the electrical wiring. I gambled around on the zwave shutter device and found no solution. In my installation I have a cut off relay which need to work with 10 other shutters. All shutters can be controlled from a centralized switch or individually . Each blind has it's own shutter device.
![Trennrelais](https://user-images.githubusercontent.com/90320300/188688605-58ab28f8-7809-4247-a54e-061abc4ff543.jpg)

Even I change the the motor side and the blind is working like expected with the inverted blind mode but all external momentary switches are wrong, or the external momentary switches are working well and the device control in the GUI is wrong. 
I do not understand what has changed because in older versions it worked like expected with the inverted blind mode. I have another installation running with Fibaro shutters also in inverted mode and no problems at all(but not the newest beta). Honestly I am totally lost

**Comment by gizmocuz** [comment]
@SargonofAssyria 

> if you open the blinds, does it say 'open' in the status page ? NO change, not even in the slider.
> if you close the blinds, does it say 'close' in the status page ? Change in 'Closed', also the slider goes to the end. See pic.
> if you set the slider to 30%, does it display 30% in the status page ? Change in 30% and slider goes to 30%
> 

Maybe because your blinds where already open it did not change, so when you press the close button then the open there should be a change (and visa versa)

A closed blind (MQTT-AD) should have the slider position at 0, as mentioned before 0-close, 99=open, so your screenshot is perfect
When you set the slider to 10%, the blind should be 10% open

Open:
![image](https://user-images.githubusercontent.com/2265299/188792800-6ed18896-f643-4aba-89dd-e2cf0482f70e.png)

Closed:
![image](https://user-images.githubusercontent.com/2265299/188792825-a287af0a-2ebd-4233-b8ea-a01d6cf33d8e.png)

To see the status in the event editor you have to open the status page:
![image](https://user-images.githubusercontent.com/2265299/188792956-6582230d-f7f7-4fde-bbe9-004042b16f0e.png)

(When the state changed (because you pressed a open/close button, you need to press refresh at the top right)
I just verified and the status updates/displays correctly in the overview, meaning you should be able to query the correct status via dzVents
(You do not need to call a JSON query for this...)

I did some debugging with dzVents and noticed some issues that have been corrected in beta 14498

I verified that opening/closing and setting a position worked for both normal and inverted blinds

I used the following small script for testing

```
local TEST_SWITCH = 7939
local MyBlind = 8232

return {
	on = {
		devices = {
			TEST_SWITCH
		}
	},
	logging = {
		level = domoticz.LOG_DEBUG,
		marker = 'MQTT-AD Blind test',
	},
	execute = function(domoticz, item)
	    domoticz.log(domoticz.devices(MyBlind).name .. ' state = ' .. domoticz.devices(MyBlind).state, domoticz.LOG_DEBUG)
            domoticz.log(domoticz.devices(MyBlind).name .. ' level = ' .. domoticz.devices(MyBlind).level, domoticz.LOG_DEBUG)

            if (domoticz.devices(TEST_SWITCH).state == "On") then
                if (domoticz.devices(MyBlind).level ~= 0) then
                    domoticz.log(domoticz.devices(MyBlind).name .. ' is not Closed, closing', domoticz.LOG_DEBUG)
                    domoticz.devices(MyBlind).close()
                    --domoticz.devices(MyBlind).setLevel(30)
                end
            else
                if (domoticz.devices(MyBlind).level ~= 100) then
                    domoticz.log(domoticz.devices(MyBlind).name .. ' is not Open, opening', domoticz.LOG_DEBUG)
                    domoticz.devices(MyBlind).open()
                end
            end
	end
}
```

Hope all is working correctly now

@Thaui57, 
I'm confused again.... "Qubino shutters" but now you are talking about a Jarolift relay.... 
As far as I know you connect your blinds directly to it (if your blinds run on 230V or 24VDC)
Are you also talking about your blinds in combination with MQTT Auto discovery or not ?
If yes, also try the new beta as it might be solved now

If not and you do not use the blind in combination with zwave/zigbee/mqtt-ad, did it ever worked before?

**Comment by Thaui57** [comment]
sorry for the confusion. This Jorolift device is only part of my installation and can not be re-wired. For better understanding the issue I made a wiring diagram. I am using Qubino shutter with ZwaveJS2Mqtt. Automatic control via DZvents and manual by momentary switches. The GUI in Domoticz is working but showing the wrong status. I can change the motor output or the switch input, 
![Qubino](https://user-images.githubusercontent.com/90320300/188830044-24f7da1e-37c9-485f-8496-a58e3e565ba6.jpg)

**Comment by SargonofAssyria** [comment]
Upgraded my test env. with MQTT-AD to 14498. I have my zigbee2mqtt 'invert_cover' explicitly set to false: open=100,close=0 just to be sure that it is inline with your settings. I use your script as a 'real' dzVents script, but it does not behave as expected. I have to investigate more and come back here.

**Comment by Thaui57** [comment]
@gizmocuz did another re-wiring this morning. Swapped the I1/I2 input and also the motor output Q1/Q2. The zwaveJS2Mqtt control panel is working like defined. Up/Down switches are ok and closed =0 and open =99. So from this end everything like requested.(before re-wiring it was inverted). Having a look on the Domoticz GUI. Blinds are not inverted like requested. below we have the results.
![image](https://user-images.githubusercontent.com/90320300/188842040-fb8fc92e-16c7-4d7c-8cdf-21151205704b.png)
Blind is open. Stutus GUI is closed. Domoticz log: Level 99% Closed

![image](https://user-images.githubusercontent.com/90320300/188843193-1d22643f-ec87-4518-b293-1e01f30252c7.png)
Blind is 80% open(20%closed). Stutus GUI is 80%. Domoticz log: Level 80% Set Level: 80 %

![image](https://user-images.githubusercontent.com/90320300/188844097-03b4f120-e4d2-408e-8b76-8603adbc58ab.png)
Blind is 20% open(nearly fully closed). Stutus GUI is 20%. Domoticz log: Level 20% Set Level: 20 %

![image](https://user-images.githubusercontent.com/90320300/188844768-bfcee5dd-d1cf-491e-b546-31dfbdd349b6.png)
Blind is closed. Stutus GUI is open. Domoticz log: Level 0% open

Switches opening and closing are working. Open = left switch, Close = right switch
All data are related to 
Version: 2022.1 (build 14495)
Build Hash: 4a8aa6ed6
Compile Date: 2022-09-01 17:10:07
dzVents Version: 3.1.8

this behavior is totally different from the past versions. I am afraid if I update the system with my Fibaro switches I have the same mess like here. Also I have to rewrite all my shutter Dzvents scripts because the level is inverted and also the on/off command.

**Comment by gizmocuz** [comment]
@Thaui57 , before continuing here, install Home Assistant and test with this as I am very sure the behaviour in Domoticz is not the same and correct.
Open=100, Close=0, 20% is blinds are 20% open (so 80% closed)
So please use HA to test first (or HA in combination with a ESPHome example)

**Comment by Thaui57** [comment]
@gizmocuz just for fun I changed the blind to Blinds percentage inverted and here we have the result
![image](https://user-images.githubusercontent.com/90320300/188855424-2b789130-3396-48e4-ba12-1cabef5d2044.png)
Blind is Open, Status GUI is Open, Domoticz log: aktueller Level 99% Open

![image](https://user-images.githubusercontent.com/90320300/188856165-0da1e4a1-f1ab-417d-8221-df2607b3f20a.png)
Blind is 20% closed, GUI Status is  20%, Domoticz log: aktueller Level 21% Set Level: 21 %

![image](https://user-images.githubusercontent.com/90320300/188856874-f81927e8-6664-47d4-8840-d3156312e10b.png)
Blind is Closed, Status GUI is Closed, Domoticz log: aktueller Level 0% Closed
Open and closing switch in the correct way.
Is that how it should work? The only question is if the slider bar is indicating the right position. If Open is fully blue the bar should be at 80% when the bling is 20% closed.

**Comment by gizmocuz** [comment]
@Thaui57 , yes this is how it should work (for normal blinds, not inverted), but maybe because of the wiring you need to use the inverted.
The only difference between normal/inverted is that the slider positions are inverted

if open = 100 and close = 0, then setting the slider at 80% means that the blind is 80% open (and this 20% closed)

What I do not understand is that you say your blind is at 20% but the gui displays 20% and I see above 21%
Could be a rounding issue, domoticz scales it to 0-100%, as your blind has 0-99 this could mean there is somewhere on some point a possible rounding issue
But this can be neglected

So it seems all is working fine ?

**Comment by SargonofAssyria** [comment]
Up front, I am using 14489 on PROD with the older version of the Stas plugin. Everything is working fine as it should with the exception that zigbee2mqtt internal is set position to 70 where my the GUI state 30% My dzVents scripts work with 30% So the 'internal' 70 does not bother me.
My TEST system is a docker on 14498 with MQTT-AD. Everything done on the Prod SYSTEM is **not** reflected on the GUI of the TEST system.
Open and close and 30% on the TEST GUI works as it should on the device and the GUI of PROD and only the 30% is ok displayed on TEST but as 70% on PROD 
The GUI on TEST always shows closed when closed is used once even is open is used next. SO after a close or 30% an open does not change the GUI, it stays on 'Closed' or 30% Every time the GUI on PROD follows the moves on the TEST system.
Now for the dzVents script.
The slider is at 30%
`2022-09-08 09:52:39.566 Status: dzVents: Info: Handling events for: "Dummy", value: "Off"
2022-09-08 09:52:39.567 Status: dzVents: Info: MQTT-AD Blind test: ------ Start external script: test.lua: Device: "Dummy (Dummy)", Index: 6
2022-09-08 09:52:39.567 Status: dzVents: Debug: MQTT-AD Blind test: Processing device-adapter for RolgordijnOverloop: Switch device adapter
2022-09-08 09:52:39.567 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop state = Closed
2022-09-08 09:52:39.567 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop level = 0
2022-09-08 09:52:39.568 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop is not Open, opening
2022-09-08 09:52:39.568 Status: dzVents: Debug: MQTT-AD Blind test: Constructed timed-command: Open
2022-09-08 09:52:39.568 Status: dzVents: Info: MQTT-AD Blind test: ------ Finished test.lua
`
GUI does NOT change, rolgordijn opens as reflected in PROD.
The next action with dzVents:
`2022-09-08 09:55:41.814 Status: dzVents: Info: Handling events for: "Dummy", value: "On"
2022-09-08 09:55:41.815 Status: dzVents: Info: MQTT-AD Blind test: ------ Start external script: test.lua: Device: "Dummy (Dummy)", Index: 6
2022-09-08 09:55:41.815 Status: dzVents: Debug: MQTT-AD Blind test: Processing device-adapter for RolgordijnOverloop: Switch device adapter
2022-09-08 09:55:41.815 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop state = Closed
2022-09-08 09:55:41.815 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop level = 0
2022-09-08 09:55:41.816 Status: dzVents: Info: MQTT-AD Blind test: ------ Finished test.lua`
No move of the rolgordijn and both the GUI's
Every other click On/Off on the dummy device gives no actions on both GUI's and the rolgordijn itself.
`2022-09-08 09:58:17.611 Status: dzVents: Info: Handling events for: "Dummy", value: "On"
2022-09-08 09:58:17.611 Status: dzVents: Info: MQTT-AD Blind test: ------ Start external script: test.lua: Device: "Dummy (Dummy)", Index: 6
2022-09-08 09:58:17.612 Status: dzVents: Debug: MQTT-AD Blind test: Processing device-adapter for RolgordijnOverloop: Switch device adapter
2022-09-08 09:58:17.612 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop state = Closed
2022-09-08 09:58:17.612 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop level = 0
2022-09-08 09:58:17.612 Status: dzVents: Info: MQTT-AD Blind test: ------ Finished test.lua
2022-09-08 09:58:18.420 Status: dzVents: Info: Handling events for: "Dummy", value: "Off"
2022-09-08 09:58:18.420 Status: dzVents: Info: MQTT-AD Blind test: ------ Start external script: test.lua: Device: "Dummy (Dummy)", Index: 6
2022-09-08 09:58:18.421 Status: dzVents: Debug: MQTT-AD Blind test: Processing device-adapter for RolgordijnOverloop: Switch device adapter
2022-09-08 09:58:18.421 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop state = Closed
2022-09-08 09:58:18.421 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop level = 0
2022-09-08 09:58:18.421 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop is not Open, opening
2022-09-08 09:58:18.421 Status: dzVents: Debug: MQTT-AD Blind test: Constructed timed-command: Open
2022-09-08 09:58:18.421 Status: dzVents: Info: MQTT-AD Blind test: ------ Finished test.lua`

What doe you want me to test next?

**Comment by gizmocuz** [comment]
@SargonofAssyria , same for you, please install HA and compare HA and Domoticz with zigbee2mqtt first before continuing

**Comment by Thaui57** [comment]
@gizmocuz I have changed the wiring on the device "Kueche" that it conforms with the ZwaveJS requirements (see comments from AlCalzone from ZwaveJS). Pls. don't care on the 20% because I guess it was 21%. So before I agree everything is working fine pls. let me know under which condition I should check the functionality? "Percentage" or "Percentage inverted"? I can confirm that many changes which has been done in the last betas are the opposite from what we had in OpenZwave. I guess many installations will suffer under this changes while changing to MQTT. Qubino has the same logic like Fibaro shutters. So no matter which shutter I am using. I have updated to 14498 today!

**_[AlCalzone] from ZwaveJS wrote on Feb. 9th 2021
I have a couple of Fibaro Roller Shutters and they work perfectly and are not inverted (0 = closed, 99 = open). The commands are guaranteed the same for both devices. The Window Covering CC specs (I know, different one, but they are designed to be compatible) describe:
Level Change Up = Opening, Level Change Down = Closing
The term “Closed” represents the lowest light throughput, while “Open” represents the highest light
throughput of the window covering._**

**Comment by SargonofAssyria** [comment]
Strange that I have to install a competitor, but I found an old image of my HA install from may this year and installed it.
The Open and Close functions exactly as my PROD Domoticz. The level settings are opposite, so 30% in Domoticz is 70% in HA, but that was to be expected. But GUI's, both in HA and Domoticz, are synchronized if one or the other is clicked for open or close or if slider is set to some value.

**Comment by SargonofAssyria** [comment]
There are 2 other 'options/attributes' on this device: reverse_direction and invert_cover both Boolean.
My current setting is both 'false' I tried all 4 combinations with Domoticz and HA, but only both settings false gave the result that both Domoticz and HA are synchronized correctly except for level setting.
B.t.w. I also have that 1 % deviation on the level setting, but I correct that with dzVents.

**Comment by gizmocuz** [comment]
The reason for asking is that for others blinds are now correctly working in Domoticz.
My ESPHome blinds sketch/hardware behaves the same in Domoticz and HA
In the latest beta versions of Domoticz the Inverted blinds should invert only the level slider and the open/close behaves the same for both
So to recap, when you use the GUI you are not able to synchronize Domoticz and HA?

Because previously you mentioned:

> when you press the open icon (left icon) does the blind open ? YES
> when you press the close icon (right icon) does the blind clsoe ? YES
> when you set the slider to 30%, does the blind open 30% (so 70% of the blind is closed) YES
> The actual display is now correct on my TEST docker 14495 environment.
> So it seems to work oke.
> 

If controlled with the GUI is OK, what type of blind have you selected? Normal of Inverted?

I tested the open/close behaviour with the script above and it does open/close, also setting the level.
When you call open/close (with my test script) does that work ?
When you set a level does that work?

Here is another user using Fibaro that seems to be working, but there seems to be a state label mismatch that has to be looked at

https://github.com/domoticz/domoticz/issues/5288#issuecomment-1240609930

**Comment by Rouzax** [comment]
Since you pointed to my comment 😄, I have not tested dzvents at this time yet, only the GUI

**Comment by SargonofAssyria** [comment]
In HA the device is exactly working as it should including the right setting of the level. And all the buttons and slider work in HA also as expected. So that is a 100% hit. My PROD system (at 14498) also works good and synchronizes with HA, except from level setting which is inverted. Both system (HA and PROD) are from one to the other correctly synchronized.
On ALL systems (HA, PROD and test MQTT-AD) the device looks like this on MQTT-AD:
![09 09 2022_10 51 29_REC](https://user-images.githubusercontent.com/24824113/189311591-4f1c52b4-5975-4592-bdb4-e466eed1665b.png)
To me that looks good. Closing the 'RolgordijnOverloop' in HA shows this in Domoticz.
![09 09 2022_10 54 13_REC](https://user-images.githubusercontent.com/24824113/189312147-fb6e5919-1dd8-41dc-bfe5-c55921824e85.png)
This looks good too.
Setting the slider to 70% in HA, does NOT change anything in the GUI of MQTT-AD (my PROD changes to 30% as expected)
All the movements are correctly executed on the hardware device it self.
Now a click on the 'Open' button on MQTT-AD GUI opens the hardware device and changes PROD and HA accordingly, but the GUI in MQTT-AD still is at the previous (Closed) display.
Now a click on the 'Close' button on the MQTT-AD GUI closes the hardware device and changes PROD and HA accordingly, and of course the GUI in MQTT-AD is still 'Closed'
Now setting the slider to 70% in MQTT-AD i shown and executed correctly on ALL systems.

![09 09 2022_11 04 41_REC](https://user-images.githubusercontent.com/24824113/189314133-1f435858-f7cb-437c-9fc3-2e6875f402db.png)

Come back in a few minutes with the results of the dzVents script.

**Comment by SargonofAssyria** [comment]
dzVents, starting point the same as above, so 'Open' GUI from MQTT-AD is still at 70% a click on Open in that GUI does NOT change anything.
Click on 'Dummy' device to 'Off' does NOT change anything in the GUI of MQTT-AD. The device opens and that is refelected in PROD en HA.
`2022-09-09 11:41:25.420 Status: dzVents: Info: Handling events for: "Dummy", value: "Off"
2022-09-09 11:41:25.421 Status: dzVents: Info: MQTT-AD Blind test: ------ Start external script: test.lua: Device: "Dummy (Dummy)", Index: 6
2022-09-09 11:41:25.421 Status: dzVents: Debug: MQTT-AD Blind test: Processing device-adapter for RolgordijnOverloop: Switch device adapter
2022-09-09 11:41:25.422 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop state = Closed
2022-09-09 11:41:25.422 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop level = 0
2022-09-09 11:41:25.422 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop is not Open, opening
2022-09-09 11:41:25.422 Status: dzVents: Debug: MQTT-AD Blind test: Constructed timed-command: Open
2022-09-09 11:41:25.422 Status: dzVents: Info: MQTT-AD Blind test: ------ Finished test.lua`
A second click on the Dummy device -> ON does not do anything.
`2022-09-09 11:45:23.892 Status: User: Admin (IP: 192.168.2.47) initiated a switch command (6/Dummy/On)
2022-09-09 11:45:23.980 Status: dzVents: Info: Handling events for: "Dummy", value: "On"
2022-09-09 11:45:23.981 Status: dzVents: Info: MQTT-AD Blind test: ------ Start external script: test.lua: Device: "Dummy (Dummy)", Index: 6
2022-09-09 11:45:23.981 Status: dzVents: Debug: MQTT-AD Blind test: Processing device-adapter for RolgordijnOverloop: Switch device adapter
2022-09-09 11:45:23.982 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop state = Closed
2022-09-09 11:45:23.982 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop level = 0
2022-09-09 11:45:23.982 Status: dzVents: Info: MQTT-AD Blind test: ------ Finished test.lua`
The real state of the device is 'Open' eg. 100%
If I want to set the level to 45% then I have to remove the 'if (domoticz.devices(MyBlind).level ~= 0) then' check.
Both PROD en HA reflect the change to 45%, but the MQTT-AD GUI stays at 70% eg. NO change.
A subsequent click to 'Off' on the Dummy opens the blinds on PROD and HA , but No change in the GUI of MQTT-AD
`2022-09-09 11:56:29.128 Status: dzVents: Info: Handling events for: "Dummy", value: "Off"
2022-09-09 11:56:29.128 Status: dzVents: Info: MQTT-AD Blind test: ------ Start external script: test.lua: Device: "Dummy (Dummy)", Index: 6
2022-09-09 11:56:29.129 Status: dzVents: Debug: MQTT-AD Blind test: Processing device-adapter for RolgordijnOverloop: Switch device adapter
2022-09-09 11:56:29.129 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop state = Closed
2022-09-09 11:56:29.129 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop level = 0
2022-09-09 11:56:29.129 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop is not Open, opening
2022-09-09 11:56:29.129 Status: dzVents: Debug: MQTT-AD Blind test: Constructed timed-command: Open
2022-09-09 11:56:29.129 Status: dzVents: Info: MQTT-AD Blind test: ------ Finished test.lua`

@gizmocuz Hope you have enough now, otherwise please ask.

**Comment by Rouzax** [comment]
@SargonofAssyria It's better to put the code between code block Markdown with 3 backticks upfront and 3 after, that helps with readability

` ``` `
paste log
` ``` `

That would give something like this

> zVents, starting point the same as above, so 'Open' GUI from MQTT-AD is still at 70% a click on Open in that GUI does NOT change anything. 
> 
> Click on 'Dummy' device to 'Off' does NOT change anything in the GUI of MQTT-AD. The device opens and that is refelected in PROD en HA.
> ```
> 2022-09-09 11:41:25.420 Status: dzVents: Info: Handling events for: "Dummy", value: "Off" 
> 2022-09-09 11:41:25.421 Status: dzVents: Info: MQTT-AD Blind test: ------ Start external script: test.lua: Device: "Dummy (Dummy)", Index: 6 
> 2022-09-09 11:41:25.421 Status: dzVents: Debug: MQTT-AD Blind test: Processing device-adapter for RolgordijnOverloop: Switch device adapter 
> 2022-09-09 11:41:25.422 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop state = Closed 
> 2022-09-09 11:41:25.422 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop level = 0 
> 2022-09-09 11:41:25.422 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop is not Open, opening 
> 2022-09-09 11:41:25.422 Status: dzVents: Debug: MQTT-AD Blind test: Constructed timed-command: Open 
> 2022-09-09 11:41:25.422 Status: dzVents: Info: MQTT-AD Blind test: ------ Finished test.lua
> ```
> 
> A second click on the Dummy device -> ON does not do anything.
> ```
> 2022-09-09 11:45:23.892 Status: User: Admin (IP: 192.168.2.47) initiated a switch command (6/Dummy/On) 
> 2022-09-09 11:45:23.980 Status: dzVents: Info: Handling events for: "Dummy", value: "On" 
> 2022-09-09 11:45:23.981 Status: dzVents: Info: MQTT-AD Blind test: ------ Start external script: test.lua: Device: "Dummy (Dummy)", Index: 6 
> 2022-09-09 11:45:23.981 Status: dzVents: Debug: MQTT-AD Blind test: Processing device-adapter for RolgordijnOverloop: Switch device adapter 
> 2022-09-09 11:45:23.982 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop state = Closed 
> 2022-09-09 11:45:23.982 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop level = 0 
> 2022-09-09 11:45:23.982 Status: dzVents: Info: MQTT-AD Blind test: ------ Finished test.lua
> ```
> 
> The real state of the device is 'Open' eg. 100% 
> If I want to set the level to 45% then I have to remove the 'if (domoticz.devices(MyBlind).level ~= 0) then' check. 
> Both PROD en HA reflect the change to 45%, but the MQTT-AD GUI stays at 70% eg. NO change. 
> A subsequent click to 'Off' on the Dummy opens the blinds on PROD and HA , but No change in the GUI of MQTT-AD
> ```
> 2022-09-09 11:56:29.128 Status: dzVents: Info: Handling events for: "Dummy", value: "Off" 
> 2022-09-09 11:56:29.128 Status: dzVents: Info: MQTT-AD Blind test: ------ Start external script: test.lua: Device: "Dummy (Dummy)", Index: 6 
> 2022-09-09 11:56:29.129 Status: dzVents: Debug: MQTT-AD Blind test: Processing device-adapter for RolgordijnOverloop: Switch device adapter 
> 2022-09-09 11:56:29.129 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop state = Closed 
> 2022-09-09 11:56:29.129 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop level = 0 
> 2022-09-09 11:56:29.129 Status: dzVents: Debug: MQTT-AD Blind test: RolgordijnOverloop is not Open, opening 
> 2022-09-09 11:56:29.129 Status: dzVents: Debug: MQTT-AD Blind test: Constructed timed-command: Open 
> 2022-09-09 11:56:29.129 Status: dzVents: Info: MQTT-AD Blind test: ------ Finished test.lua
> ```
>

**Comment by SargonofAssyria** [comment]
Tnx, now hope for the solution....

**Comment by SargonofAssyria** [comment]
@gizmocuz Is there anything I could test to make things working? Because build 14498 is working fine with my Production env. and OLD zigbee plugin from Stas, and my TEST env. with latest zigbee plugin (v 3.10-beta) from Stas is working fine too.
But MQTT-AD is not working correctly. This is one of the last devices preventing me to switch to MQTT-AD in Production.

**Comment by kiddigital** [comment]
@SargonofAssyria , there are a few tests you can do to help gathering some details.

Keep the following separation in mind:
- the hardware module that talks to the actual devices (like MQTT-AD, the plugin, Zigbee, etc) and stores its data into the Domoticz database (incl a hardware type reference)
- the Domoticz UI that gets it data, including the hardware type reference, from the database and displays it.

To get some more insights, please look at the data being stored in the database (tables) and the data send to the UI (use the developer console of the browser to see which data is exchanged with the UI). Would be great to learn where/what the difference are between your setups.

Thx for your valuable time.

**Comment by SargonofAssyria** [comment]
I looked at the console of my Docker MQTT-AD env. Every action on the other systems is shown on there consoles but not on my Docker MQTT-AD Lights and PIR actions are shown in the console but not the actions from the blinds.
1 thing should be added that the slider of my TEST env. with the latest Stas plugin does not react on the percentage. Position value (like 30%) in the upper right corner is correct but the slider not.

**Comment by SargonofAssyria** [comment]
In my TEST env. (normal Domoticz) I deleted ALL zigbee devices (deleted hardware) and recreated the devices from the Stas latest plugin and the MQTT-AD After done that this is the result.
![19 09 2022_17 20 20_REC](https://user-images.githubusercontent.com/24824113/191053522-c302d3ca-5f70-415f-9ff2-24fb2c779fb8.png)
The upper row is from the Stas plugin and the lower row is from MQTT-AD The upper row, except for the slider, is correct with percentages, the MQTT-AD row does NOT reflect the real status of the blinds.
This is the database entry of RolgordijnOverloop
![19 09 2022_17 29 18_REC](https://user-images.githubusercontent.com/24824113/191055228-de7a4efa-0e0d-48e2-b498-c87e2cf02e7a.png)

**Comment by ksacca** [comment]
@SargonofAssyria I didn’t really see your issue, even though we have a similar problem (for me it looked like dzvents-related according to the title). 
I was able to fix a similar issue today (blinds not updating, always shown as closed). This is also for zigbee, no issues with zwave shutters. 

See https://www.domoticz.com/forum/viewtopic.php?p=292630#p292630 for more info (don’t mind the previous post, I was completely wrong there). 

In short, I believe Domoticz uses the value_template when available instead of the position_template. This points to the state attribute instead of the position one (hence no state change in domoticz). If you remove the value template, for me it works as it should. Hope this will fix your issue.

**Comment by SargonofAssyria** [comment]
Initially I tought there where 2 problems. GUI and dzVents. @gizmocuz asked me to change the title only to dzVents problem. You can see above. I then found out that the GUI problem still exists, so I started yesterday to investigate with fresh installs etc. because there was no progress. I started yesterday with just the GUI problem. The results are right above. I implemented your solution with:
`    homeassistant:
      value_template: null
`
and that worked for my PROD en TEST env. with Stas. plugin and MQTT-AD and also in HA. So kudos for you!
So now I will test all the dzVents scenarios and come back here with the results.
Your workaround is not the final solution. So question remains will/should it be resolved in Domoticz or Zigbee2mqtt?

**Comment by SargonofAssyria** [comment]
I have to correct myself. In my TEST env. with the latest Stas. plugin everything but the slider works correct. The slider does not move. I tested the dzVents script from @gizmocuz and that works correctly with MQTT-AD

**Comment by ksacca** [comment]
I don't use the Stas plugin (https://github.com/stas-demydiuk/domoticz-zigbee2mqtt-plugin), but I installed it to test what's happening on your side. In the latest version of the stas plugin something changed for blinds (invert blind on/off behavior, see https://github.com/stas-demydiuk/domoticz-zigbee2mqtt-plugin/blob/master/CHANGELOG.md). 

With the latest version of the stats plugin, the slider doesn't move. But the value is updated.

![Recording 2022-09-22 at 10 58 12](https://user-images.githubusercontent.com/9539267/191704872-e0047451-8499-4816-8fd6-f01de8b854f7.gif)

But if you revert to the previous version (v 3.0) using `git reset --hard f8c22fc330229b39f06ff11bbfcea38e3a492306`, the slider works (althoug reversed, as it was fixed in the next version I suppose). So it looks like the slider not moving is rather a plugin issue.

![Recording 2022-09-22 at 10 55 05](https://user-images.githubusercontent.com/9539267/191703745-5649c4e2-9075-48ec-a551-b34cf12cf6e7.gif)

For me it still looks like removing the "value_template" from the discovery payload is still the right solution. It's just that's there another issue with the plugin you use. It's up to @gizmocuz to see how he could be able to solve the "value_template" in domoticz, or if it should be fixed in zigbee2mqtt. I'm not smart enough to know where the issue comes from 😄 .

**Comment by SargonofAssyria** [comment]
I never used and will never use the 'new' version of the Stas plugin because of the way the 254 device limit is handled with the api* device. So for me the choice is MQTT-AD or my current solution in PROD with the 'old' Stas plugin. So I agree with you that the solution has to come from Domoticz or Z2M. My preference is Domoticz, because HA works fine with "value_template" included. I can/will only switch to MQTT-AD when all my devices work as they work now in PROD with the 'old' Stas plugin. And currently it is NOT.

**Comment by gizmocuz** [comment]
I am currently a bit busy, maybe you could try to help with the c++ development ?

**Comment by rezzalex** [comment]
It is way beyond my skills, but why not give it a try... I would need some step by step beginner guide for the dev environment.

**Comment by gizmocuz** [comment]
@rezzalex , please start by using visual studio community edition as it's easy to set breakpoints
There is some documentation in the INSTALL.md file

**Comment by gizmocuz** [comment]
@ksacca , are you able to debug/modify the code ?
I looked at the code, but on line 2825 (MQTTAutoDiscovery.cpp) i do check if the 'position' field is found and use this

```
		if (!root["position"].empty())
		{

```

**Comment by gizmocuz** [comment]
There have been made some recent changes that might solve this issue
Also do not mix non-mqtt auto discovery blinds with blinds from other hardware as they do behave differently by level
In MQTT-AD , open = 100 and close=0 where for other hardware (have always been like this) open=0 and close=100

**Comment by ksacca** [comment]
I just installed Version: 2022.1 (build 14534), but I still have to remove the remove the value_template from zigbee2mqtt for the blinds to update in domoticz, so that hasn't changed. The issue now is that the state of the blind is reversed (open when closed, closed when open).

build 14534:
![build 14534](https://user-images.githubusercontent.com/9539267/194089210-6bb31bb9-c819-4d03-bab9-285833aaf57b.gif)

build 14498:
![build 14498](https://user-images.githubusercontent.com/9539267/194090326-09c0eff2-ccbd-4062-b4d8-9dcee72d81f8.gif)

Since this system is still a test to migrate to MQTT autodiscovery, I quickly plugged in my zwave stick. With ZwaveJS UI, the states of the blind are correct. So just to be clear, both zwave and zigbee blinds are created by the MQTT with autodiscovery hardware.

**Comment by gizmocuz** [comment]
@ksacca , the final open/close state reported in the widget i see... And you are not using a inverted blind ?

**Comment by gizmocuz** [comment]
Just to be sure please clear your browser cache also

**Comment by ksacca** [comment]
It is indeed the domoticz widget that is reversed. As you can see in the images, it's a regular blind percentage (not percentage inverted) but setting it to percentage inverted doesn't change the behaviour.

Zigbee2mqtt has an option to invert blinds, but when I set this, my changes to the value_template are always overwritten, so the state in domoticz doesn't change and I'm not able to test it.
![image](https://user-images.githubusercontent.com/9539267/194100183-833e71fc-4e09-4e1e-a562-97afc68a4fa3.png)


Just tried again after clearing browser cache as instructed in the wiki on https://www.domoticz.com/wiki/Browser_Cache, but the domoticz widget still shows the blinds are closed, while they are actually open.
![image](https://user-images.githubusercontent.com/9539267/194099833-c258cf04-f9fc-4894-8ea8-6bc153c23525.png)

Blind fully closed (but open according to domoticz) gets nvalue 1:

```
2022-10-05 17:35:27.987 Debug: zigbee: Topic: zigbee2mqtt/Gordijn, Message: {"battery":83,"linkquality":0,"position":0,"state":"CLOSE","update":{"state":"idle"},"update_available":false}
2022-10-05 17:35:27.989 Debug: SQLH UpdateValueInt Gordijn battery HwID:6 DevID:0x0c4314fffe3fc90a_battery_zigbee2mqtt Type:243 sType:6 nValue:0 sValue:{"battery":83,"linkquality":0,"position":0,"state":"CLOSE","update":{"state":"idle"},"update_available":false}
2022-10-05 17:35:27.999 Debug: SQLH HandleOnOffAction: OnAction:
2022-10-05 17:35:27.999 Debug: SQLH UpdateValueInt Gordijn (MQTT AD) HwID:6 DevID:0x0c4314fffe3fc90a_cover_zigbee2mqtt Type:244 sType:73 nValue:1 sValue:100
```

Blind fully opened (but closed according to domoticz) gets nvalue 0:

```
2022-10-05 17:37:51.843 Debug: zigbee: Topic: zigbee2mqtt/Gordijn, Message: {"battery":83,"linkquality":0,"position":100,"state":"OPEN","update":{"state":"idle"},"update_available":false}
2022-10-05 17:37:51.844 Debug: SQLH UpdateValueInt Gordijn battery HwID:6 DevID:0x0c4314fffe3fc90a_battery_zigbee2mqtt Type:243 sType:6 nValue:0 sValue:{"battery":83,"linkquality":0,"position":100,"state":"OPEN","update":{"state":"idle"},"update_available":false}
2022-10-05 17:37:51.855 Debug: SQLH HandleOnOffAction: OffAction:
2022-10-05 17:37:51.855 Debug: SQLH UpdateValueInt Gordijn (MQTT AD) HwID:6 DevID:0x0c4314fffe3fc90a_cover_zigbee2mqtt Type:244 sType:73 nValue:0 sValue:0
```

**Comment by gizmocuz** [comment]
Working on a PR that should correct it all

https://github.com/domoticz/domoticz/pull/5361

Feel free to test when/if possible, after compilation and starting make sure to clear the browser cache as there are changes in the GUI

**Comment by gizmocuz** [comment]
Quite more work went into this PR... I think it is ready for testing...

**Comment by gizmocuz** [comment]
The PR has been merged and there are no Inverted blinds anymore...
When you edit a switch/blind, you now have the option to reverse the state and/or position via checkboxes
Beta 14536 will be available within the hour for all platforms (lots of changes, lots to build)
Do not forget to clear your browser cache as the change involves some GUI changes/code as well!

**Comment by Thaui57** [comment]
**Version: 2022.1 (build 14536) with Zwave-JS-UI version 8.1 and DZvents**
Bravisimo, Congratulation and many thanks. After many unsuccessful attempts I can confirm that the GUI is now acting like the last stable. I had given up to continuously check beta versions and report malfunction because nothing changed. But now it seems to work. thanks for your efforts. Let's see if my dzvents scripting is also working well.

**Comment by Thaui57** [comment]
**Please don't forget the timer functionallity**
I am running my device with the reverse state= on
![image](https://user-images.githubusercontent.com/90320300/194708240-39b45b80-5484-417b-bac9-d3392a54ed49.png)
So if you are using the timer for the blind **OFF ** is closing the blind and **ON** is opening the blind. In case that you are using the reverse state you never can open the blind to fully open because it's limited to 1%. In my case the window is crashing with the roller shutter.

**Comment by Rouzax** [comment]
There seems to be an issue indeed with the conversion from 0-100 of Domoticz to the 0-99 from Zwave with this build. I'm documenting everything at the moment and will send that to @gizmocuz 

With the timer that might also be an issue as it starts from 1 not 0, not sure yet how the Off and On behave there.
![image](https://user-images.githubusercontent.com/4103090/194709097-0d6be5fa-0b8a-4696-addb-3898ca9b1fc6.png)

**Comment by gizmocuz** [comment]
The issue from 1-99 as well as possible gui slider notation issues have been solved in beta 14541
Timers are indeed remaining...

**Comment by gizmocuz** [comment]
@Thaui57, Is it possible to try beta 14542 for the Timers? Hope they work now and are clearer😁
Not sure what you mean by 'In my case the window is crashing with the roller shutter'
You can either select Open (and choose a level from 1-100) or select Close (which will obviously use 0 as position internally)

**Comment by Rouzax** [comment]
> @Thaui57, Is it possible to try beta 14542 for the Timers? Hope they work now and are clearer😁 Not sure what you mean by 'In my case the window is crashing with the roller shutter' You can either select Open (and choose a level from 1-100) or select Close (which will obviously use 0 as position internally)

What he means, if the blind stays at 1% instead of 0% it might still be a bit before the window and if your windows open to the outside it will crash against the blind. That was the result of the bug you already fixed

**Comment by Thaui57** [comment]
@gizmocuz  I have tested beta 14542. the only thing I found out that I have now an open/close pulldown. Sorry but this is not sufficient. If I use the timer (no open script) I can open only to 1%. My balcony door is opening von inner to outer position. So if the timer has opened the blind in the morning and the position is remaining at 1% the door is crashing with the blind. I hope that make the request a little clearer.

**Comment by gizmocuz** [comment]
@Thaui57 , You also have the open/close in the list of timers
The range from 1-100 has always been here.
Open=100 and Close=0. If you want to send the value 0, you need to program the timer with the 'Close' command

**Comment by gizmocuz** [comment]
@Thaui57 , please do note that blind behaviour has changed to match other hardware/software.

Previously:

Open=0 , Close = 100

Actual:

Open=100, Close = 0

**Comment by ksacca** [comment]
Thank you very much for your hard work @gizmocuz! I was not able to test yet as I can't get the docker containers working since beta 14535. I always get an error in the log:

```
/usr/local/bin/docker-entrypoint.sh: line 35: /opt/domoticz/domoticz: cannot execute binary file: Exec format error
/usr/local/bin/docker-entrypoint.sh: line 35: /opt/domoticz/domoticz: Success
```

I notice some changes were made in this file yesterday, and some other changes were pushed today regarding dockerized builder tools. I'll try again if a new build is available.

**Comment by gizmocuz** [comment]
@ksacca , that's correct, I merged some changes in the docker build process, maybe they are the cause of the issue.
I will try to revert them

**Comment by gizmocuz** [comment]
I have reverted the changes and build a new docker, is it possible to check if that is working again ?

**Comment by ksacca** [comment]
Thanks a lot again. Just pulled beta 14543 and confirm the docker image works again. I'll make sure to test all different blinds/shutters I have available (dummy blinds using GPIO relay, zwave blinds with fibaro roller shutter (2 & 3) and one using zigbee2mqtt. I'll report if I encounter any issues.

**Comment by Thaui57** [comment]
@gizmocuz in the current beta it's not possible to achieve a fully open blind if the switch is inverted. Closing means closing and percentage has no functionality. 100% open means a closed blind if it's inverted. In any case it will remain a 1% open blind if you chose 1% open.Am 09.10.2022 13:14 schrieb gizmocuz ***@***.***>:
@Thaui57 , You also have the open/close in the list of timers
The range from 1-100 has always been here.
Open=100 and Close=0. If you want to send the value 0, you need to program the timer with the 'Close' command

**Comment by gizmocuz** [comment]
@Thaui57  no idea what you say or if you said anything
Do not reply to emails when using GitHub but directly respond in GitHub please

**Comment by Thaui57** [comment]
@gizmocuz okay maybe my request is unclear described. I run all my blinds with a reverse open/close state.
![image](https://user-images.githubusercontent.com/90320300/195179213-3701df43-31d8-4a7c-80fd-5ebf821ceb8d.png)
In this situation 0= Open and 100= closed (see also the response from Rouzax 3 day ago)
Under this conditions I have no possibility to fully open the blind and in any case a 1% opening will still remain. Maybe it makes also sense to have 3 options in the pulldown. Open - Close and Percentage (1-99%)

**Comment by Rouzax** [comment]
The device limit is automatically recalculated in Domoticz to 100, so for my Fibaro devices which also go from 0 to 99 this does work

**Comment by gizmocuz** [comment]
@Thaui57 , I am not 100% following what you say

You display a screenshot from 'edit device' where you can enable the reverse options

Indeed, when you reverse the state, and you click the 'open' icon in the widget, 0 is send to the actual device
When you click the 'close' icon, 100 is send to the actual device

Maybe a better understanding of the workings of the blinds is explained in this post:

https://github.com/domoticz/domoticz/issues/5359#issuecomment-1274347465

**Note** , if you are using ZWaveJS-UI or Zigbee2MQTT you probably do not need to reverse anything if you blind goes up/down
Also make sure you did not enable any reverse/inverted option in the GUI of ZWaveJS-UI or Zigbee2MQTT

Next you mention 1-99, are you talking about the slider in the widget?

Then you talk about a pulldown? In the 'Edit Device' option, there is no pulldown for open/close
Are you talking about a scene/group maybe?
If you do, under 'normal' conditions when you choose 'Close' it will send 0, and when you select 'Open' you can choose a percentage from 1-100. So you see, you can have the option from 0-100

Question remains
- Why do you need to reverse the Open/Close state
- Don't you need to reverse the position as well?
- What type of blinds are you using

**Comment by Thaui57** [comment]
@Rouzax I don't need 100, I need a 0 for opening the blind to 100%
@gizmocuz yes I have changed from OZW to Zwave-JS-UI and you are right normally I do not need a reversed blind with MQTT. But my installation has worked with reversed blinds in the past because of electrical wiring. So I need to have a 0 to fully open the blind and 100(99) to close it.
![image](https://user-images.githubusercontent.com/90320300/195278823-a4155a34-af84-4a67-a753-191a3a195387.png)
100 is no problem but how to achieve the 0 for open. I am talking about the timer and not about scripting.
![image](https://user-images.githubusercontent.com/90320300/195280189-f8682c94-0a62-40e3-84c9-8acb40aaf484.png)
in timer you have the command for open and close as a pulldown. Close is closing the blind(percentage has no functionallity), Open you have the range from 1-99% and opening the blind. I do need a 0% or an opening command. 
I am using Qubino ZMNHCD. I hope this is clarify the situation for better understanding.
I need to achieve the old behavior with the new beta
Previously:
Open=0 , Close = 100

**Comment by Rouzax** [comment]
Then enable both checkboxes in the new settings

**Comment by gizmocuz** [comment]
@Thaui57 , thanks for the clarification, sure we could not have known that you were talking about timers.
As mentioned before, close does have a meaning, this is 0.

So what you need is to reverse both the state and percentage

Reverse Open/Close State: ✔️
Reverse Position: ✔️
This will reverse the State Label, and will reverse the Slider position

- 0 = Blind Open in GUI/dzvents
- 100 = Blind Closed in GUI/dzvents
- 90 = Show 90 in the GUI/dzvents, Send 10 to the device
- 10 = Show 10 in the GUI/dzvents, Send 90 to the device

So try this first, set your blinds to a percentage (30%) and add a timer in three minutes that has 'close'
see if that gives you the desired behavior
Next add a timer in again three minutes and set it to 30%, see if it goes to the previous 30%
If that works, add a timer in three minutes and set it to 100%
You should see that you can use the full 0-100 range

**Comment by gizmocuz** [comment]
https://www.domoticz.com/wiki/Blinds

**Comment by Thaui57** [comment]
@gizmocuz @Rouzax thanks for the advise. This did the trick, but now I have to change all my percentages in the dzvents scripts. However if there will be no better solution we have to live with it or I use  a script as a workaround. This was the reason why I have mentioned to edit the pulldown with 3 options. Open/Close/Percentage.

**Comment by gizmocuz** [comment]
Script? I thought you were talking about timers…
In your script you also call .open .close and the set level just as the timer or gui
If you have to invert everything in your script and you invert everything in the switch itself, maybe you should not invert to begin with
But great it’s working for you too

**Comment by gizmocuz** [comment]
@Thaui57 , could it be that the reason for the need to change the percentages in your scripts is because the logic in Domoticz has changed (and is now the same as in other software) ? Like 0=Close and 100=Open ?
In this case yes, you need to change your scripts

**Comment by Thaui57** [comment]
@gizmocuz you are right. I run timers and in parallel some dzvents scripts for shading. I will find a solution for my systems no doubt but very logical is the existing situation not really.  To enter 100% to achieve 0%. I guess many questions will come up after the next stable. The rest of the changes are working perfect now.Thanks for that.

**Comment by Rouzax** [comment]
But everything should be the same if you are using:

- Reverse Open/Close State: ✔️
- Reverse Position: ✔️

Then this will invert the Percentages and State everywhere, in GUI, dzvents and Timers.
If you want 100% to be 100% don't enable the reverse position.

Maybe it will help us understand if you created a detailed post like I did https://github.com/domoticz/domoticz/issues/5288#issuecomment-1240609930 and describe and show all the states and explain why they are off.

**Comment by gizmocuz** [comment]
Exactly, sending 100 should send 100 to the blind (unless you have reversed the position/state, then it will actually send 0 to the blind, still in Domoticz it is threaded as 100)
